### PR TITLE
TransferRequest.Payload is usually in hex string format #46

### PR DESCRIPTION
**Pull request by dustinxie**
This is for issue #46 
also remove dependency to iotex-core/action

### CODE REVIEW

File: action/action_test.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2019 IoTeX
```
Comment (dustinxie) [inline action/action_test.go]: move this file from account/ to action/ folder

File: iotx/iotx.go [hunk 2]
```diff
@@ -89,6 +89,15 @@ func (i *Iotx) SendTransfer(req *TransferRequest) (string, error) {
 		return "", err
 	}
 
+	amount, ok := big.NewInt(0).SetString(req.Value, 10)
+	if !ok {
+		return "", ErrAmount
+	}
+	data, err := hex.DecodeString(req.Payload)
```
Comment (dustinxie) [inline iotx/iotx.go]: Paylod is usually in hex string format
so should use hex.DecodeString(), but not directly convert to []byte like []byte(req.Payload)

Comment (ququzone) [reply iotx/iotx.go]: I think as a SDK, payload is plain text.

Comment (dustinxie) [reply iotx/iotx.go]: 比如我想送 "hello world" 应该设 req.Payload = "68656c6c6f20776f726c64"，而不是 req.Payload = "hello world"
输入 payload，data = hex.DecodeString(payload)，然后把 data 传给 NewTransfer 或者 NewExecution
这样的话 transfer 和 contract 的处理一致，比如网页前端要 deploy 一个contract，bytecode 也是这样的 hex string 格式

**Comment by dustinxie** [comment]
> if delete `Gopkg.lock` file, so can add to .gitignore?

this PR did not delete Gopkg.lock, it just changed as a result of removing dependency

**Pull request merged**